ln-gateway/src/messaging.rs

```diff
@@ -0,0 +1,40 @@
+use anyhow::Error;
+use async_trait::async_trait;
+use tokio::sync::{mpsc, oneshot};
+
+use crate::{GatewayRequest, GatewayRequestTrait, Result};
+
+#[async_trait]
+pub trait GatewayMessageReceiver: Send + Sync + 'static {
+    async fn receive(&self) -> ();
+}
+
+#[derive(Debug, Clone)]
+pub struct GatewayMessageChannel {
+    sender: mpsc::Sender<GatewayRequest>,
+}
+
+/// A two-way message channel for [`GatewayRequest`]s.
```

Comment:
Two way message (request + response) is an "rpc", so maybe `GatewayRpcSender`?

Reply:
BTW. "Gateway" here is not very useful. Everything is about "gateway".

`InternalRpcSender` maybe would be better? We are planning to use it for inter-actor rpcs?

Reply:
Adopting

Reply:
While at it, I'll rename this file to `rpc.rs`. I had that `messaging` name because of this struct